src/carma-1-tenth/check_message_timing.py

```diff
@@ -0,0 +1,49 @@
+# Check the times between messages received on two topics
+
+from rosbag_utils import open_bagfile
+import numpy as np
+import yaml
+import tqdm
+import datetime
+import argparse, argcomplete
+import os
+
+
+def check_message_timing(bag_dir, call_topic, response_topic):
+    metadatafile : str = os.path.join(bag_dir, "metadata.yaml")
+    if not os.path.isfile(metadatafile):
+        raise ValueError("Metadata file %s does not exist. Are you sure %s is a valid rosbag?" % (metadatafile, bag_dir))
+    with open(metadatafile, "r") as f:
+        metadata_dict : dict = yaml.load(f, Loader=yaml.SafeLoader)["rosbag2_bagfile_information"]
+    storage_id = metadata_dict['storage_identifier']
+    reader, type_map = open_bagfile(bag_dir, topics=[call_topic, response_topic], storage_id=storage_id)
+    topic_count_dict = {entry["topic_metadata"]["name"] : entry["message_count"] for entry in metadata_dict["topics_with_message_count"]}
+    if call_topic not in topic_count_dict or response_topic not in topic_count_dict:
+        return np.array([np.inf])
+    call_topic_count, response_topic_count = 0,0
+    call_topic_times, response_topic_times = np.zeros((topic_count_dict[call_topic],)), np.zeros((topic_count_dict[response_topic],))
+    # Iterate through bag and store odometry + route_graph messages
+    for _ in tqdm.tqdm(iterable=range(topic_count_dict[call_topic] + topic_count_dict[response_topic])):
+        if(reader.has_next()):
+            (topic, data, t_) = reader.read_next()
```

Comment:
t_ feels like a weird variable name. Some of the "r"s and "f"s are standard enough to be ok but let's avoid single variable characters to the degree possible

Reply:
Changed to `timestamp`